rules/rbe_repo.bzl

```diff
@@ -718,91 +702,169 @@ def rbe_autoconfig(
-    if not digest:
-        fail(("Could not find a valid digest for revision %s, " +
-              "please make sure it is declared in " +
-              "@bazel_toolchains//rules:toolchain_containers.bzl" % revision))
-
-    # If the user has set a custom env, custom java_home,
-    # registry or repository, don't use
-    # checked in configs
-    if ((env and env != clang_env()) or
-        (java_home) or
-        (registry and registry != _RBE_UBUNTU_REGISTRY) or
-        (repository and repository != _RBE_UBUNTU_REPO)):
-        use_checked_in_confs = False
+
+    if bazel_rc_version and not bazel_version:
+        fail("bazel_rc_version can only be used with bazel_version.")
+
+    # Resolve the Bazel version to use.
+    if not bazel_version or bazel_version == "local":
+        bazel_version = str(extract_version_number(_BAZEL_VERSION_FALLBACK))
+        rc = parse_rc(native.bazel_version)
+        bazel_rc_version = rc if rc != -1 else None
+
+    if tag and digest:
+        fail("'tag' and 'digest' cannot be set at the same time.")
+
+    if not ((not digest and not tag and not repository and not registry) or
+            (digest and repository and registry) or
+            (tag and repository and registry)):
+        fail("All of 'digest', 'repository' and 'registry' or " +
+             "all of 'tag', 'repository' and 'registry' or " +
+             "none of them must be set.")
+
+    # Set to defaults only if all are unset.
+    if not repository and not registry and not tag and not digest:
+        repository = _RBE_UBUNTU_REPO
+        registry = _RBE_UBUNTU_REGISTRY
+        tag = RBE_UBUNTU16_04_LATEST
+
+    if registry and registry == _RBE_UBUNTU_REGISTRY and repository and repository == _RBE_UBUNTU_REPO and not env:
+        env = clang_env()
+
+    resolved_config_info = validateUseOfCheckedInConfigs(
+        registry,
+        repository,
+        tag,
+        digest,
+        use_checked_in_confs,
+        env,
+        java_home,
+        bazel_version,
+        bazel_rc_version,
+    )
+
+    use_checked_in_confs = False
+    digest = digest
+    tag = tag
+    config_version = None
+    if resolved_config_info:
+        use_checked_in_confs = True
+        digest = resolved_config_info.digest
+        tag = resolved_config_info.tag
+        config_version = resolved_config_info.config_version
+
     _rbe_autoconfig(
         name = name,
         bazel_version = bazel_version,
-        bazel_rc = bazel_rc,
+        bazel_rc_version = bazel_rc_version,
         config_dir = config_dir,
+        config_version = config_version,
         digest = digest,
         env = env,
         exec_compatible_with = exec_compatible_with,
         java_home = java_home,
         output_base = output_base,
         registry = registry,
         repository = repository,
-        revision = revision,
+        tag = tag,
         target_compatible_with = target_compatible_with,
         use_checked_in_confs = use_checked_in_confs,
```

Comment:
I think you dont need to pass down use_checked_in_confs and you can just check if config_version is not None

Reply:
Done.